### ISSUE TITLE
Token(basic) Classes lack operators overloading

### ISSUE THREAD

**Issue by Sekgobela-Kevin** [Sekgobela-Kevin/cnavy#6]
**Basic_Token**, **Basic_Characters** and **Basic_Tokeniser** objects lack overloaded operators. It would be much easy to access their data with operators other than having to call method. Overloading of [], ==, + and << is neccessary to improve efficiency when working with basic tokens.

**Comment by Sekgobela-Kevin** [comment]
### That was challenging.

- I had to refactor some classes and templates, also update tests to match latest changes.
- Templates do the impossible but comes with responsibilities.
- Rafactoring Raw_Tokenise template class was the hardest.
- Tests could could only help after finishing.
- It takes lot of compiler errors to get right

But at the end of the day, use of C++ programming improves.

**Comment by Sekgobela-Kevin** [comment]
This marks the first time escaping from **Character** and **Characters** objects and transitioning to **Raw_Character,** **Basic_Token** and **Basic_Tokeniser** which opens opportunity to work with **XTess** **Componets**.

**Comment by Sekgobela-Kevin** [comment]
I have been waiting to get here for more than a month. 
I cant believe I got here with no experience of _C/C++_.

Without this project, I couldnt have leant all the things I have learnt. Programming is more than just reading or attending a course but needs practice, patience and accepting to failure as opportunity to grow. 

That Google IT Automation with Python course was worth as it. I know use git through command line with Bash which is efficient than GUI version. I have used Python for years but I wasnt expecting write any working program with C++.

Hope the the project succeed and perform better than the Python version(private).

Pynavy is coming soon with code from my private repo I worked last year which was hosted in [www.datanavy.site](www.datanavy.site) but know its down after several months. It was using Spacy and required picking which was also using lot of disk space. It was also slow to create doc object and searching was also slow due to algorithm even if it was optimised multiple times.

CNavy is expected beat Datanavy search.
